src/main/java/com/puppycrawl/tools/checkstyle/DetailAstImpl.java

```diff
@@ -52,54 +51,58 @@
     /** Previous sibling. */
     private DetailAstImpl previousSibling;
 
+    /** First child of this DetailAST. */
+    private DetailAstImpl firstChild;
+
+    /** First sibling of this DetailAST.*/
+    private DetailAstImpl nextSibling;
+
+    /** Text of this DetailAST. */
+    private String text;
+
+    /** The type of this DetailAST. */
+    private int type;
+
+    /**
+     * All tokens on COMMENTS channel to the left of the current token up to the
+     * preceding token on the DEFAULT_TOKEN_CHANNEL.
+     */
+    private List<Token> hiddenBefore;
```

Comment:
Many of the changes here were part of the ANTLR2 'AST' interface that we relied on https://www.antlr2.org/javadoc/antlr/collections/AST.html.